### PR TITLE
refactor(wordpress): make wp-config-ddev.php a static asset, fixes #8098

### LINKED ISSUE #8098: Improve WordPress support, especially generated files
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Is your feature request related to a problem?

DDEV's WordPress support is not very sophisticated and can be improved significantly, even though there are so many different permutations of WP layout and config.

### Describe your solution

For a start, use DDEV_PRIMARY_URL for WP_HOME.

- [ ] In wp-config-ddev.php, we shouldn't be putting the literal URL for WP_HOME, it should be $DDEV_PRIMARY_URL instead, so 
   ```
    $wp_home = getenv('DDEV_SHARE_URL') ?: getenv('DDEV_PRIMARY_URL');                                                                                              
  defined('WP_HOME') || define('WP_HOME', $wp_home);           
    ```

### Describe alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by rfay**
## The Issue

- Fixes #8098

`wp-config-ddev.php` was generated via Go's `text/template`, injecting values like DB credentials and host at creation time. All those values are either DDEV constants (`'db'`) or available as runtime env vars (`DDEV_PRIMARY_URL`), so the template was unnecessary complexity.

This follows on from #8176, and incorporates the review suggestion from https://github.com/ddev/ddev/pull/8176#discussion_r3035542219 to rename `$wp_home` → `$ddev_wp_home` to avoid conflicts with plugins or snippets that define a global `$wp_home`.

## How This PR Solves The Issue

- Converts `wordpress/wp-config-ddev.php` from a Go template to a static file: hardcodes `'db'` for DB name/user/password/host (consistent with how Drupal does it), uses `getenv('DDEV_PRIMARY_URL')` for `WP_HOME`, renames `$wp_home` to `$ddev_wp_home`, and derives `WP_SITEURL` directly from `WP_HOME`.
- Updates `writeWordpressDdevSettingsFile` to use `bundledAssets.ReadFile` + `os.WriteFile` instead of template parsing, matching the pattern used for Drupal's static `settings.php`.
- Removes 8 now-unused fields from `WordpressConfig` (`WPGeneric`, `DatabaseName`, `DatabaseUsername`, `DatabasePassword`, `DatabaseHost`, `Docroot`, `TablePrefix`, `Signature`) and drops the `*DdevApp` parameter from `NewWordpressConfig`.

## Manual Testing Instructions

1. `ddev config --project-type=wordpress && ddev start`
2. Confirm generated `wp-config-ddev.php` contains `DDEV_PRIMARY_URL` and hardcoded `'db'` credentials
3. Verify WordPress loads correctly

## Automated Testing Overview

- `TestWriteSettings` updated to check for `DDEV_PRIMARY_URL` in the generated file

## Release/Deployment Notes

- Existing `#ddev-generated` `wp-config-ddev.php` files will be overwritten on next `ddev start` with the new static content
- No behavior change for user-managed `wp-config-ddev.php` files

### CODE REVIEW

**Review by jonesrussell — COMMENTED** [review]
**On `pkg/ddevapp/wordpress/wp-config-ddev.php`, the `WP_SITEURL` line:**

Nice cleanup overall — the static asset approach is much simpler and the `$ddev_wp_home` rename makes sense.

One question: the old template had:

```php
defined( 'WP_SITEURL' ) || define( 'WP_SITEURL', WP_HOME . '/{{ $config.AbsPath }}' );
```

which would append the subdirectory (e.g. `wp/`) when WordPress core lives in a subfolder of the docroot. The new version sets `WP_SITEURL` to just `WP_HOME`:

```php
defined( 'WP_SITEURL' ) || define( 'WP_SITEURL', WP_HOME );
```

Is that intentional? For subdirectory installs (where `wordpressGetRelativeAbsPath` returns something like `"wp"`), this would change `WP_SITEURL` from `https://mysite.ddev.site/wp` to `https://mysite.ddev.site`, which could break those setups.

*Review by Claude Code on behalf of @jonesrussell*

**Comment by rfay** [comment]
That's probably valid problem. I think this goes clear back to
* https://github.com/ddev/ddev/pull/932
* https://github.com/ddev/ddev/issues/1167

**Comment by stasadev** [comment]
I pushed another commit to fix `WP_SITEURL` and used `getenv()` in more places, so people can override more settings with environment variables instead of editing PHP files.

About the subdirectory and `WP_SITEURL`:

If I understand correctly, testing follows this method: https://developer.wordpress.org/advanced-administration/server/wordpress-in-directory/#method-ii-with-url-change

@jonesrussell, is that correct? Do you think we could add automated testing for subdirectory to `docs/tests/wordpress.bats`? We already have "WordPress wp-cli based quickstart (different docroot)" there, but it didn't fail on @rfay’s first change, so I assume we’re not testing this fully - or maybe we're testing only docroot, not moving WP to a subdirectory.

File: pkg/ddevapp/wordpress/wp-config.php [hunk 1]
```diff
@@ -30,11 +30,11 @@
 /* That's all, stop editing! Happy publishing. */
 
 /** Absolute path to the WordPress directory. */
-defined( 'ABSPATH' ) || define( 'ABSPATH', dirname( __FILE__ ) . '/{{ $config.AbsPath }}' );
+defined( 'ABSPATH' ) || define( 'ABSPATH', __DIR__ . '/{{ $config.AbsPath }}' );
```
Comment (stasadev) [inline pkg/ddevapp/wordpress/wp-config.php:33]: BTW, it's already defined in the https://github.com/WordPress/WordPress/blob/master/wp-load.php#L21, which is loaded before this file.

I don't know the history of `ABSPATH`, but it doesn't harm to have it here.

**Comment by jonesrussell** [comment]
I checked docs/tests/wordpress.bats: the existing “different docroot” case only tests a non-root docroot, not the WordPress Method II “subdirectory with URL change” flow. So, it would not have caught the original WP_SITEURL regression.

I drafted a dedicated Bats case for that scenario locally: install WordPress at the root, move core files into a subdirectory, copy index.php back to the root, then assert WP_SITEURL and the root/admin URLs behave correctly.

I’ll run that test once I’ve got the full Bats/DDEV test environment set up, and if it passes I’ll push it into this PR.

**Comment by jonesrussell** [comment]
Update: got the environment set up and the test passes. Branch with the new bats case is here: https://github.com/jonesrussell/ddev/tree/20260407_jonesrussell_test_wordpress_subdirectory_url

One thing worth noting: with the current template-based wp-config-ddev.php, WP_SITEURL is computed at ddev start time — so after moving files to the subdirectory, a ddev restart is needed to trigger a rescan and regenerate the file with the correct path. I added that to the test.

With the runtime ABSPATH-based approach in this PR, that restart may no longer be needed — the URL would resolve correctly based on wherever ABSPATH points. Happy to update the test to match once this lands.

**Comment by stasadev** [comment]
Thank you!

> Happy to update the test to match once this lands.

@jonesrussell, no new PR is needed since we're adding a new test for the changed code and just need to confirm it works.

I'll test your bats on @rfay's commit, then on my commit to confirm fail/success.

After that, I'll push your commit here and handle the newly found edge case with `ddev config --docroot=./wordpress` gracefully (my code didn't handle `.` or `./`).

You'll still keep the commit credit.

File: pkg/ddevapp/wordpress/wp-config.php (hunk 1 above)
Comment (stasadev) [reply pkg/ddevapp/wordpress/wp-config.php:33]: No, it's fine to have this logic for `ABSPATH`

**Comment by jonesrussell** [comment]
My bad on that PR. I'm using sonnet on extra usage. It's proving to be much less capable than Opus.
